### PR TITLE
feat(minor): add legacy (uppercase) chain name support

### PR DESCRIPTION
**Pull request by cgorenflo**

### CODE REVIEW

File: packages/router-api/src/primitives.rs [hunk 1]
```diff
@@ -191,15 +183,47 @@ impl TryFrom<String> for ChainName {
...
+    fn from(value: &ChainNameRaw) -> Self {
+        ChainName(value.0.to_lowercase())
+    }
+}
+
+impl From<ChainNameRaw> for ChainName {
+    fn from(value: ChainNameRaw) -> Self {
+        ChainName(value.0.to_lowercase())
+    }
+}
```
Comment (cjcobb23) [inline packages/router-api/src/primitives.rs:204]: Do we need these? I feel like these could be dangerous.

File: contracts/gateway/src/state.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@ pub(crate) enum Error {
 const CONFIG_NAME: &str = "config";
 const CONFIG: Item<Config> = Item::new(CONFIG_NAME);
 const OUTGOING_MESSAGES_NAME: &str = "outgoing_messages";
-const OUTGOING_MESSAGES: Map<CrossChainId, Message> = Map::new(OUTGOING_MESSAGES_NAME);
+const OUTGOING_MESSAGES: Map<&CrossChainId, Message> = Map::new(OUTGOING_MESSAGES_NAME);
```
Comment (cjcobb23) [inline contracts/gateway/src/state.rs:56]: will this need a migration?

**Review by cjcobb23 — APPROVED** [review]
We should probably add tests somewhere that ensure routing works as we expect:

1. Mixed case chain names from core are preserved.
2. Cannot route two different versions of the same message by just changing the case of the chain name.

Not sure where these should go though.

File: contracts/gateway/src/state.rs (hunk 2 above)
Comment (cgorenflo) [reply contracts/gateway/src/state.rs:56]: no, the data is identical

**Pull request merged**